lib/src/internal/pubsub_client.dart

```diff
@@ -61,20 +61,29 @@ class ClientPubsub implements AbstractPubsubClient {
   }
 
   @override
-  TopicSubscribeResponse subscribe(String cacheName, String topicName,
-      {Int64? resumeAtTopicSequenceNumber}) {
+  Future<TopicSubscribeResponse> subscribe(String cacheName, String topicName,
+      {Int64? resumeAtTopicSequenceNumber}) async {
     var request = SubscriptionRequest_();
     request.cacheName = cacheName;
     request.topic = topicName;
     request.resumeAtTopicSequenceNumber =
         resumeAtTopicSequenceNumber ?? Int64(0);
     try {
       var stream = _grpcManager.client.subscribe(request);
-      return TopicSubscription(stream, request.resumeAtTopicSequenceNumber,
-          this, cacheName, topicName);
+      final subscription = TopicSubscription(stream,
+          request.resumeAtTopicSequenceNumber, this, cacheName, topicName);
+
+      try {
+        await subscription.init();
+        return subscription;
+      } catch (e) {
+        rethrow;
+      }
```

Comment:
had to make subscribe an async function in order to run the init method that checks for the first message being a heartbeat